internal/util.go

```diff
@@ -71,12 +71,17 @@ func GetResourceLabel(resource client.Object, labelName string) (string, error)
 	return labelValue, nil
 }
 
-func DefaultCacheOptions() cache.Options {
+func DefaultCacheOptions(namespaces []string) cache.Options {
+	defaultNamespaces := map[string]cache.Config{}
+	for _, v := range namespaces {
+		defaultNamespaces[v] = cache.Config{}
+	}
 	return cache.Options{
 		ByObject: map[client.Object]cache.ByObject{
 			&apicorev1.Secret{}: {
 				Label: k8slabels.Everything(),
 			},
 		},
+		DefaultNamespaces: defaultNamespaces,
```

Comment:
What happens if the `namespaces` param is empty? Do we need this cache option then?

Reply:
ok, should be fine if passing an empty map to `DefaultNamespaces`